docs/how-to-integrate-a-frontend/how-to-integrate-dotwallet.md

```diff
@@ -0,0 +1,196 @@
+  
+- [https://dotwallet-example.vercel.app/callback/](https://dotwallet-example.vercel.app/callback/) is the redirection URI used for product deployment.
+- [http://localhost:3000/callback/](http://localhost:3000/callback/) is the redirection URI used for development and debugging.
+
+![](../../static/img/DotWallet-uris.png)
+
+
+:::note
+*Callback domain* in the form is the redirection URIs in OAuth. 
+:::
+
+## Implementation
+
+
+
+Here is a code example to create `DotWalletSigner` in [Nextjs](https://nextjs.org/):
+
+
+1. Construct URI. 
+
+```ts
+export default async function Home() {
+  const loginUrl = `https://api.ddpurse.com/authorize?client_id=${process.env.CLIENT_ID}&redirect_uri=${encodeURIComponent(
+      process.env.REDIRECT_URI || ''
+    )}&response_type=code&scope=${encodeURIComponent(
+      "user.info"
+    )}&state=${crypto.randomUUID()}`;
+  
+    return (
+      <main className="flex min-h-screen flex-col items-center justify-between p-24">
+        <div className="m-4 p-4 bg-blue-200 font-bold rounded-lg">
+          <a href={loginUrl}>DotWallet Login</a>
+        </div>
+      </main>
+    );
+}
+```
+
+<center>src/app/page.tsx</center>
+
+
+If the user clicks the **DotWallet Login** button, the page will be redirected to the authorization server page.
+
+![](../../static/img/DotWallet-auth.png)
+
+
+2. After the user clicks **Agree to authorize** to log in, the authorization server redirect the user to the redirection URI. The following code receives the `code` through the callback uri. Then exchange the `code` for `access_token` and save it.
+
+```ts
+import { redirect, notFound } from 'next/navigation';
+
+import token from "../token"
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const code = searchParams.get('code');
+
+  if (code) {
+    // exchange the code for access_token
+    const res = await fetch(`https://api.ddpurse.com/v1/oauth2/get_access_token`, {
+      body: JSON.stringify({
+        code,
+        redirect_uri: process.env.REDIRECT_URI,
+        grant_type: "authorization_code",
+        client_secret: process.env.CLIENT_SECRET,
+        client_id: process.env.CLIENT_ID,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST'
+    });
+    const { code: apiCode, data, msg } = await res.json();
+
+    if (apiCode === 0) {
+      const { access_token } = data;
+      // save access_token
+      token.access_token = access_token;
+      // redirect to balance page.
+      redirect('/balance');
+    }
+
+  }
+
+  notFound();
+}
+```
+
+<center>src/app/callback/route.ts</center>
```

Comment:
`route.ts` is `http://localhost:3000/callback/`? Explain how callback is routed at here. It is not clear.

Reply:
Inside the [app directory](https://nextjs.org/docs/app/building-your-application/routing/defining-routes#creating-routes), folders are used to define routes in nextjs. we create `src/app/callback/route.ts` to handle the redirection request.